torch/csrc/jit/tracer.cpp

```diff
@@ -408,6 +437,11 @@ void TracingState::setValue(const IValue& v, Value* value) {
     for (size_t i = 0; i < elements.size(); ++i) {
       setValue(elements[i], unpack_node->outputs()[i]);
     }
+  } else if (
+      v.isObject() && v.toObject()->type()->name() &&
```

Comment:
seems that in 3 places we're doing a multi-step unwrap that boils down to `bool isCustomClass(const IValue&)`—can we turn it into a utility function defined in `custom_class.h`